smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -633,6 +646,23 @@ private void generateErrorDeserializer(GenerationContext context, StructureShape
         writer.write("");
     }
 
+    private List<HttpBinding> readErrorResponseBody(
+            GenerationContext context,
+            Shape error,
+            HttpBindingIndex bindingIndex,
+            boolean isBodyParsed
+    ) {
+        TypeScriptWriter writer = context.getWriter();
+        if (isBodyParsed) {
+            // Body is already parsed in error dispatcher, simply assign body to data.
+            writer.write("const data: any = output.body;");
+            return ListUtils.of();
```

Comment:
This will short circuit deserializing any modeled content in the document of an error response. There is no call to `deserializeOutputDocument` and any shapes that may need to be deserialized [won't be added to the `deserializingDocumentShapes` set here](https://github.com/awslabs/smithy-typescript/pull/66/files#diff-d253d30297eab6ee7abb02d06e3c223eR639).